packages/openauth/src/storage/sqlite.ts

```diff
@@ -0,0 +1,59 @@
+import { joinKey, splitKey, type StorageAdapter } from "./storage.js";
+import Database from "libsql";
+
+export interface SqliteStorageOptions {
+  persist?: string;
+  tableName?: string;
+}
+
+export function SQLiteStorage(input?: SqliteStorageOptions): StorageAdapter {
+  // initialize sqlite database and create the necessary table structure
+  const db = new Database(input?.persist ?? ":memory:");
+  const TABLE_NAME = input?.tableName ?? "__openauth__kv_storage";
+
+  db.exec(
+    `CREATE TABLE IF NOT EXISTS ${TABLE_NAME} (key TEXT PRIMARY KEY, value TEXT, expiry INTEGER)`
```

Comment:
expiry is a separate column instead of being part of the value, as it's relatively easier to query.